remix/app/root.tsx

```diff
@@ -57,9 +66,39 @@ export function CatchBoundary() {
   );
 }
 
+export const action: ActionFunction = async ({ request }) => {
+  await auth.logout(request, { redirectTo: '/login' });
+};
+
+type LoaderData = {
+  viewer?: any;
+  valid?: any;
+};
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const session = await sessionStorage.getSession(
+    request.headers.get('Cookie')
+  );
+  const valid = session.get(auth.sessionKey);
+
+  // verifies authed before calling to prevent redirect spiral
+  if (valid) {
+    const { accessToken } = await auth.isAuthenticated(request, {
+      failureRedirect: '/login',
+    });
+    const { viewer } = await gqlClient.request(CURRENT_USER_QUERY, undefined, {
+      authorization: `Bearer ${accessToken}`,
+    });
+    return json<LoaderData>({ viewer, valid });
+  }
```

Comment:
Not needed but if we want to refactor and clean up a bit more this is a nice pattern for the session https://github.com/kentcdodds/kentcdodds.com/blob/main/app/utils/session.server.ts#L57 - Could just have a method on the session value returned from getSession to handle checking auth value etc

Reply:
I'll make a separate enhancement ticket in case we want to go this route.

Reply:
https://github.com/thisdot/starter.dev-github-showcases/issues/251

Reply:
We should def due #251 since Kent recommends it.